sentry-ruby/lib/sentry/scope.rb

```diff
@@ -47,6 +48,69 @@ def dup
       copy
     end
 
+    def set_user(user_hash)
+      check_argument_type!(user_hash, Hash)
+      @user = user_hash
+    end
+
+    def set_extras(extras_hash)
+      check_argument_type!(extras_hash, Hash)
+      @extra = extras_hash
+    end
+
+    def set_extra(key, value)
+      @extra.merge!(key => value)
+    end
+
+    def set_tags(tags_hash)
+      check_argument_type!(tags_hash, Hash)
+      @tags = tags_hash
+    end
+
+    def set_tag(key, value)
+      @tags.merge!(key => value)
+    end
+
+    def set_contexts(contexts_hash)
+      check_argument_type!(contexts_hash, Hash)
+      @contexts = contexts_hash
```

Comment:
Hi! is there a specific reason this method overwrites `@contexts` instead of merging? Tags and extras (which are conceptually similar) are both merged

Reply:
I believe this is a bug. thanks for raising this, I'll fix it 🙂